api/app/Console/Jobs/RandomizeTextJob.php

```diff
@@ -1,181 +1,203 @@
+        
+        /**
+         * @var bool is logging enabled.
+         */
+        protected bool $loggingEnabled = true;
+        
+        protected bool $timeoutEnabled = false;
+        
+        protected int $maxJobTime = 60 * 60;
+        
+        private XmlGeneration $xmlGeneration;
+        
+        /**
+         * Randomises text in specified result column based on pattern column and updates spreadsheet.
+         *
+         * @param string $tableId spreadsheet id.
+         * @param int $patternCol column to take pattern from.
+         * @param int $resultCol column to fill in randomized result.
+         * @param int $numRow row number, required for spreadsheet update.
+         * @param string $sheetName sheet name, required for spreadsheet update.
+         * @param array $row row data.
+         * @param string $quotaUser quota user.
+         * @throws \Exception
+         */
+        private function randomizeText(
+            string $tableId,
+            int $patternCol,
+            int $resultCol,
+            int $numRow,
+            array $row,
+            string $sheetName,
+            string $quotaUser): void
         {
-            return;
-        }
-
-        $this->log("Randomizing row ".$numRow);
-
-        $text = $this->spintaxService->randomize($row[$patternCol]);
-
-        // Счет строк начинается с 1, а не с 0 и первая строка - заголовок
-        $numRow += +2;
-        $columnName = SpreadsheetHelper::getColumnLetterByNumber($resultCol);
-        $this->spreadsheetClientService->updateCellContent(
-            $tableId, $sheetName, $columnName.$numRow, $text, $quotaUser);
-    }
-
-
-    /**
-     * Randomize text for specified generator.
-     *
-     * @param string $tableID Google spreadsheet id.
-     * @param string $sheetName sheet name.
-     * @param string $quotaUserPrefix quota user prefix.
-     */
-    private function processSheet(string $tableID, string $sheetName, string $quotaUserPrefix): void
-    {
-        [ $propertyColumns, $values ] = $this->getHeaderAndDataFromTable($tableID, $sheetName, $quotaUserPrefix);
-
-        if (empty($values))
-        {
-            return;
+            $alreadyFilled = isset($row[$resultCol]) && $row[$resultCol] != '';
+            $noSource = !isset($row[$patternCol]) || $row[$patternCol] == '';
+            
+            if ($alreadyFilled || $noSource) {
+                return;
+            }
+    
+            // Счет строк начинается с 1, а не с 0 и первая строка - заголовок
+            $numRow += +2;
+            
+            $message = "Table '".$tableId."' randomizing row ".$numRow."...";
+            $this->log($message);
+            Log::info($message);
+            
+            $text = $this->spintaxService->randomize($row[$patternCol]);
+            
+            try {
+                $columnName = SpreadsheetHelper::getColumnLetterByNumber($resultCol);
+                $this->spreadsheetClientService->updateCellContent(
```

Comment:
Were not you going to update the whole column at once?

Reply:
later, after base functional will be ready